### PR TITLE
Raise coverage and fix meta option completion

### PR DESCRIPTION
**Pull request by dahlia**
This PR raises the official `mise coverage` result to the target 95% line and branch coverage threshold, with focused tests across *packages/core*, *packages/env*, *packages/config*, *packages/inquirer*, *packages/git*, *packages/temporal*, *packages/valibot*, *packages/zod*, *packages/logtape*, and the *examples/gitique* example.

Most of the new coverage is in parser edge cases: replay behavior, fallback metadata, dependency replay, deferred parser state, completion probes, source extraction, and value parser normalization. The goal was not to pin implementation details, but to cover the behaviors that could break user-facing parsing, completion, or fallback flows.

While adding completion coverage, this found and fixed a shell completion bug in *packages/core/src/facade.ts*. Configured help and version options such as `--help`, `--version`, aliases, slash-prefixed names, and plus-prefixed names are now suggested at the root option position. Completion payloads such as `build --help ""` and `build --version ""` are still treated as completion context, not as active help/version requests.

This also fixes plus-prefixed option completion in *packages/core/src/primitives.ts*, so parser options and flags like `+debug` are recognized as option-prefix completions. The regression coverage lives in *packages/core/src/facade.test.ts* and *packages/core/src/primitives.test.ts*. The changelog entry is in *CHANGES.md*.

## Testing

- `mise coverage`
- `mise check && mise test:deno && CI=true mise test:node && CI=true mise test:bun`

## Coverage result

- Official processed `mise coverage` table: branch 95.0%, line 95.0%

### CODE REVIEW

**Comment by dahlia** [comment]
@codex review

File: packages/core/src/valueparser.ts [hunk 1]
```diff
@@ -5096,7 +5038,9 @@ export function macAddress(
       const finalSeparator = outputSeparator ?? inputSeparator ?? ":";
       return { success: true, value: joinOctets(octets, finalSeparator) };
     },
-    format: normalizeMac, // overridden below
+    format(value: string): string {
+      return value;
+    },
   };
   // Both format and normalize use the full parse() pipeline so that
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/valueparser.ts]: Declined: the pass-through formatter is not the effective macAddress formatter. The parser immediately replaces it with the parse-pipeline formatter via Object.defineProperty, so unnormalized defaults are already canonicalized and non-string values still fall back to the metavar.

File: packages/core/src/facade.ts [hunk 2]
```diff
@@ -1812,14 +1636,80 @@ function handleCompletion<M extends Mode, THelp, TError>(
...
+  const combined = [...suggestions];
+  for (const suggestion of extraSuggestions) {
+    if (prefix === "-" && suggestion.text.length !== 2) continue;
+    if (!suggestion.text.startsWith(prefix)) continue;
+    if (seen.has(suggestion.text)) continue;
+    combined.push(suggestion);
+    seen.add(suggestion.text);
+  }
+  return combined;
+}
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:1712]: Addressed in 004e42ec: root meta options are now suggested at empty prompts and after subcommands using the current completion prefix.

File: examples/gitique/src/index.test.ts [hunk 3]
```diff
@@ -101,4 +105,233 @@ describe("gitique CLI", () => {
...
+    };
+    assert.equal(parsed.command, "log");
+    assert.equal(parsed.format, "oneline");
+    assert.equal(parsed.maxCount, 3);
+    assert.equal(parsed.since.getFullYear(), 2024);
+    assert.equal(parsed.since.getMonth(), 0);
+    assert.equal(parsed.since.getDate(), 2);
+    assert.equal(parsed.until.getFullYear(), 2024);
+    assert.equal(parsed.until.getMonth(), 11);
+    assert.equal(parsed.until.getDate(), 31);
```
[the remark this answers is not in the document]
Comment (dahlia) [reply examples/gitique/src/index.test.ts:247]: Declined: this parser intentionally treats YYYY-MM-DD as a local calendar date using new Date(year, month - 1, day) and validates it with local date getters. Switching this assertion to UTC getters would fail in non-UTC timezones and would no longer match the parser contract.

File: packages/core/src/dependency-runtime.test.ts [hunk 4]
```diff
@@ -1165,6 +1325,126 @@ describe("extractRawInputFromState", () => {
...
+    assert.notStrictEqual(record.nested, state.nested);
+    assert.deepEqual(record.nested[0], {
+      success: true,
+      value: "warn:prod",
+    });
+    assert.deepEqual(record[symbolKey], {
+      success: true,
+      value: "warn:prod",
+    });
+  });
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/dependency-runtime.test.ts:1434]: Addressed in d22a26cc: shared deferred parse states are cached during runtime resolution so repeated references preserve result identity.

File: packages/core/src/dependency.test.ts [hunk 5]
```diff
@@ -4425,6 +4425,34 @@ describe("coverage-guided dependency parser tests", () => {
...
+      dependencies: [modeParser] as const,
+      factory: () => ({
+        mode: "sync" as const,
+        metavar: "VALUE",
+        placeholder: "ok" as const,
+        parse(_input: string): ValueParserResult<"ok"> {
+          const thenable = () => undefined;
+          thenable.then = () => undefined;
+          // @ts-expect-error: intentional function-thenable regression coverage.
+          return thenable;
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/dependency.test.ts]: Addressed in a1076b3a: the function-thenable fixture now has an explicit callable type with a then property, keeping the test fixture type-safe.

File: packages/core/src/modifiers.test.ts [hunk 6]
```diff
@@ -6854,6 +6921,172 @@ describe("branch coverage: modifiers edge cases", () => {
...
+    const result = parser.parse({
+      buffer: ["--same"],
+      state,
+      optionsTerminated: false,
+      usage: parser.usage,
+    });
+
+    assert.ok(result.success);
+    if (!result.success) return;
+    assert.equal(result.provisional, true);
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/modifiers.test.ts]: Addressed in a1076b3a: the new boolean assertions now use assert.ok instead of assert.equal(..., true).

File: packages/core/src/modifiers.test.ts [hunk 7]
```diff
@@ -9591,90 +9824,308 @@ describe(
...
+            [Symbol.for(`@test/issue-594/${name}/phase2-null-retry`)]: true,
           });
 
-          const seed = completeOrExtractPhase2Seed(parser, outerState);
+          const seed = extractPhase2Seed(parser, outerState);
 
-          assert.equal(completeCalls, 2);
+          assert.deepEqual(seed, { value: "LIVE" });
+          assert.equal(completeCalls, 1);
           assert.equal(extractCalls, 2);
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/modifiers.test.ts]: Addressed in a1076b3a: the phase-two tests now assert observable retry and unwrapping behavior instead of exact complete/extract call counts.

**Comment by dahlia** [comment]
@codex review

File: packages/core/src/dependency-runtime.test.ts [hunk 8]
```diff
@@ -361,6 +388,44 @@ function unwrappingExtract(
...
+      value: options.dependencyIds,
+    });
+  }
+  if (options.defaultValues != null) {
+    Object.defineProperty(parser, defaultValues, {
+      value: options.defaultValues,
+    });
+  }
+  return parser;
+}
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/dependency-runtime.test.ts:427]: Addressed in 4b2e06d1: the derived parser helper now relies on its explicit type annotation without a redundant type assertion.

File: packages/core/src/facade.ts [hunk 9]
```diff
@@ -927,246 +931,61 @@ interface MetaCommandGroups {
...
+    parse: () => ({
+      success: false,
+      error: message`Documentation-only meta option.`,
+      consumed: 0,
+    }),
+    complete: (state) => ({ success: true, value: state }),
+    suggest: function* () {},
+    getDocFragments: (state) => source.getDocFragments(state),
+  };
+}
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:962]: Addressed in 4b2e06d1: documentation-only meta option parsers now forward the default value and render fragments with the wrapped parser initial state.

**Comment by dahlia** [comment]
@codex review

File: packages/core/src/facade.ts [hunk 10]
```diff
@@ -1812,14 +1640,92 @@ function handleCompletion<M extends Mode, THelp, TError>(
...
+
+  const prefix = args.at(-1) ?? "";
+  const completedArgs = args.slice(0, -1);
+  const metaNames = new Set(
+    extraSuggestions.map((suggestion) => suggestion.text),
+  );
+  if (
+    completedArgs.some((arg) =>
+      metaNames.has(arg) ||
+      [...metaNames].some((name) => arg.startsWith(`${name}=`))
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts]: Addressed in adcc3fa4: the meta option equals-form check now reuses the existing suggestion list instead of rebuilding the meta-name array inside the loop.

File: packages/core/src/facade.ts [hunk 11]
```diff
@@ -1812,14 +1640,92 @@ function handleCompletion<M extends Mode, THelp, TError>(
...
+    )
+  ) {
+    return suggestions;
+  }
+  if (
+    !(prefix === "" || prefix.startsWith("--") || prefix.startsWith("-") ||
+      prefix.startsWith("/") || prefix.startsWith("+"))
+  ) {
+    return suggestions;
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:1695]: Addressed in adcc3fa4: root meta options are no longer injected while completing option value slots, including option-like value prefixes and ambiguous branch shapes.

File: packages/core/src/facade.ts [hunk 12]
```diff
@@ -1812,14 +1640,92 @@ function handleCompletion<M extends Mode, THelp, TError>(
...
+      metaNames.has(arg) ||
+      [...metaNames].some((name) => arg.startsWith(`${name}=`))
+    )
+  ) {
+    return suggestions;
+  }
+  if (
+    !(prefix === "" || prefix.startsWith("--") || prefix.startsWith("-") ||
+      prefix.startsWith("/") || prefix.startsWith("+"))
+  ) {
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:1694]: Addressed in adcc3fa4: root meta options are now suppressed after a real options terminator while preserving values that happen to be "--".

**Comment by dahlia** [comment]
@codex review

File: packages/core/src/facade.ts [hunk 13]
```diff
@@ -1808,18 +1641,413 @@ function handleCompletion<M extends Mode, THelp, TError>(
...
+
+  const metaNames = new Set(
+    extraSuggestions.map((suggestion) => suggestion.text),
+  );
+  if (
+    completedArgs.some((arg) =>
+      metaNames.has(arg) ||
+      extraSuggestions.some((suggestion) =>
+        arg.startsWith(`${suggestion.text}=`)
+      )
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts]: Addressed in bf98ed89: completion now tracks actual parser context, so option values such as --help no longer suppress later root meta suggestions.

File: packages/core/src/facade.ts [hunk 14]
```diff
@@ -1808,18 +1641,413 @@ function handleCompletion<M extends Mode, THelp, TError>(
...
+        for (const branch of term.terms) {
+          collectDuplicateCommandBranchOptionNames(
+            branch,
+            commandPath,
+            names,
+          );
+        }
+      }
+    }
+  }
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts]: Addressed in bf98ed89: active value-option names are collected for the current command context, including root and matched subcommand positions.

File: packages/core/src/facade.ts [hunk 15]
```diff
@@ -1808,18 +1641,413 @@ function handleCompletion<M extends Mode, THelp, TError>(
...
+
+  const seen = new Set(
+    suggestions.flatMap((suggestion) =>
+      suggestion.kind === "literal" ? [suggestion.text] : []
+    ),
+  );
+  const combined = [...suggestions];
+  for (const suggestion of extraSuggestions) {
+    if (prefix === "-" && suggestion.text.length !== 2) continue;
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts]: Addressed in bf98ed89: bare - completion now excludes only --long meta options, so configured single-dash aliases remain suggested.

File: packages/core/src/facade.ts [hunk 16]
```diff
@@ -1808,18 +1641,413 @@ function handleCompletion<M extends Mode, THelp, TError>(
...
+    if (prefix !== "" && !suggestion.text.startsWith(prefix)) continue;
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts]: Addressed in bf98ed89: configured single-dash meta aliases such as -help are now included after a bare - prompt.

File: packages/core/src/facade.ts [hunk 17]
```diff
@@ -1808,18 +1641,413 @@ function handleCompletion<M extends Mode, THelp, TError>(
...
+      usage,
+      context.exec?.commandPath ?? [],
+    );
+  return {
+    completingOptionValue: result.consumed > 0
+      ? valueOptionNames.has(context.buffer[0])
+      : duplicateCommandBranchValueOptionNames.has(context.buffer[0]),
+    optionsTerminated: context.optionsTerminated,
+  };
+}
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:1942]: Addressed in bf98ed89: completion value-slot detection now uses the active command context and the parser failure consumption count.

**Comment by dahlia** [comment]
@codex review

File: packages/core/src/facade.ts [hunk 18]
```diff
@@ -1808,18 +1642,368 @@ function handleCompletion<M extends Mode, THelp, TError>(
...
+
+function collectOptionNamesAtCurrentCommandDepthFromTerm(
+  term: Usage[number],
+  names: CurrentOptionNames,
+  afterMatchedCommand: boolean,
+): boolean {
+  switch (term.type) {
+    case "command":
+      return !afterMatchedCommand;
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:2114]: Addressed in b1e2f434: root value options declared after command terms are now detected while completing their values, so help and version meta options stay out of that value slot.

File: packages/core/src/facade.ts [hunk 19]
```diff
@@ -1808,18 +1642,368 @@ function handleCompletion<M extends Mode, THelp, TError>(
...
+): CompletionArgumentContext {
+  const activeValueOptionNames = collectActiveValueOptionNames(
+    usage,
+    context.exec?.commandPath ?? [],
+    result.consumed > 0,
+  );
+  const token = context.buffer[0];
+  return {
+    completingOptionValue: (result.consumed === 0 || result.consumed === 1) &&
+      activeValueOptionNames.has(token),
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:1937]: Addressed in b1e2f434: I kept the consumed == 0 path because root value options after command terms can fail before consuming the option token. That path is now limited to root value options, so unknown option tokens still receive meta alternatives.

**Comment by dahlia** [comment]
@codex review

File: packages/core/src/valueparser.ts [hunk 20]
```diff
@@ -5096,7 +5038,9 @@ export function macAddress(
       const finalSeparator = outputSeparator ?? inputSeparator ?? ":";
       return { success: true, value: joinOctets(octets, finalSeparator) };
     },
-    format: normalizeMac, // overridden below
+    format(value: string): string {
+      return value;
+    },
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/valueparser.ts:5058]: Addressed in f38d061d: format() and normalize() now use the parse-based MAC normalization path directly in the parser object, so the implementation visibly preserves the configured case and separator behavior.

**Comment by dahlia** [comment]
@codex review

File: packages/core/src/facade.ts [hunk 21]
```diff
@@ -1797,7 +1621,17 @@ function handleCompletion<M extends Mode, THelp, TError>(
     parser.mode,
     () => {
       const syncParser = parser as Parser<"sync", unknown, unknown>;
-      const suggestions = suggest(syncParser, args as [string, ...string[]]);
+      const argumentContext = getSyncCompletionArgumentContext(
+        syncParser,
+        args.slice(0, -1),
+        rootOptionSuggestions,
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts]: Addressed in ceea1d86: completion now reuses the single prefix parse for both suggestions and root meta-option gating, so async and runtime-backed parsers are not parsed twice for one completion response.

**Comment by dahlia** [comment]
@codex review

File: packages/core/src/facade.ts [hunk 22]
```diff
@@ -1808,18 +1645,532 @@ function handleCompletion<M extends Mode, THelp, TError>(
...
+  const activeValueOptionNames = collectActiveValueOptionNames(
+    parser.usage,
+    context.exec?.commandPath ?? [],
+    result.consumed > 0,
+    parser.leadingNames,
+  );
+  const token = context.buffer[0];
+  return {
+    completingOptionValue: (result.consumed === 0 || result.consumed === 1) &&
+      activeValueOptionNames.has(token),
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:1937]: Declined: consumed === 0 is required for duplicate command branches where a branch has matched the command but the active value option is rejected without consuming the option token. Existing regression coverage expects meta options to stay suppressed while completing that value slot.

File: packages/core/src/facade.ts [hunk 23]
```diff
@@ -1808,18 +1645,532 @@ function handleCompletion<M extends Mode, THelp, TError>(
...
+        result,
+        context,
+        parser,
+        rootOptionSuggestions,
+      );
+      const suggestions: Suggestion[] = [];
+      const suggestContext = await withCompletionSuggestRuntimeAsync(
+        parser,
+        context,
+      );
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:1802]: Declined: the async path follows the same ParserResult contract as the sync path. Failed parses provide consumed count and error only, so there is no updated next context to preserve for suggestion seeding.

File: packages/core/src/facade.ts [hunk 24]
```diff
@@ -1808,18 +1645,532 @@ function handleCompletion<M extends Mode, THelp, TError>(
...
+    if (!result.success) {
+      const argumentContext = getFailedCompletionArgumentContext(
+        result,
+        context,
+        parser,
+        rootOptionSuggestions,
+      );
+      return {
+        suggestions: Array.from(
+          parser.suggest(withCompletionSuggestRuntime(parser, context), prefix),
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:1755]: Declined: parser failures do not expose a next context in ParserResult, and top-level suggestSync/suggestAsync intentionally call suggest with the current context after a failed prefix parse. Using result.next would not typecheck or match the parser contract.
